### DISCUSSION TITLE
round out clipboard actions with add actions to delete/change selection to clipboard

### DISCUSSION THREAD

**Discussion by KiaraGrouwstra** [helix-editor/helix#11156] [Ideas] [answered]
i'm considering having my helix default to using the system clipboard.

this currently seems supported by the following mappings:

- `yank` -> `yank_to_clipboard`
- `replace_with_yanked` -> `replace_selections_with_clipboard`
- `paste_after` -> `paste_clipboard_after`
- `paste_before` -> `paste_clipboard_before`

however, the following actions still seem to be missing such `clipboard` equivalents:

- `delete_selection`
- `change_selection`

it would be nice to see such equivalents added as as say  `delete_selection_to_clipboard` and `change_selection_to_clipboard`.
while i am a bit new to rust, it looks like this would involve adding such functions similar to [the existing ones](https://github.com/helix-editor/helix/blob/a75b1cf51e445eb021ccd875210f529f09cd5f53/helix-term/src/commands.rs#L2745-L2759) to allow these new equivalents to perhaps override [its register](https://github.com/helix-editor/helix/blob/a75b1cf51e445eb021ccd875210f529f09cd5f53/helix-term/src/commands.rs#L2673-L2674) maybe not unlike [the yank variants](https://github.com/helix-editor/helix/blob/a75b1cf51e445eb021ccd875210f529f09cd5f53/helix-term/src/commands.rs#L4102-L4110).

**Comment by KiaraGrouwstra** [comment]
i tried to think if such actions could already be emulated using a sequence of existing actions.
however, selecting a register currently seems to only support an interactive selection, whereas operations such as `yank_to_clipboard` seem to currently call `exit_select_mode(cx);`, making it harder to tag on subsequent selection-based commands like `delete_selection_noyank` or `change_selection_noyank`.

**Reply by the-mikedavis** [reply, marked as answer]
That should work:

```toml
[keys.normal]
c = ["yank_to_clipboard", "change_selection_noyank"]
d = ["yank_to_clipboard", "delete_selection_noyank"]
```

I think you might be mixing up what `exit_select_mode` does with the `collapse_selection` or `keep_primary_selection` commands. `exit_select_mode` just goes to normal mode if you're in select mode - it doesn't affect the selections: https://github.com/helix-editor/helix/blob/fd7b1a3e3722ec49d42f795a475806c476799c30/helix-term/src/commands.rs#L3561-L3565

**Comment by KiaraGrouwstra** [comment]
nor does the [configuration](https://docs.helix-editor.com/configuration.html) seem to support specifying a default/initial register.

**Reply by the-mikedavis** [reply]
That's something that macro keybindings would cover (#1383) - you would be able to bind `@"+` to select the `+` register (using the `select_register` command). It's not currently possible to bind a key to select a register though
